internal/controller/api/http_proxy_connection_manager.go

```diff
@@ -52,82 +50,45 @@ func (rcl *RedisConnectionLocator) GetConnection(account string, node_id string)
 
 func (rcl *RedisConnectionLocator) GetConnectionsByAccount(account string) map[string]controller.Receptor {
 
-	var cursor uint64
-	var err error
-
 	connectionsPerAccount := make(map[string]controller.Receptor)
 
-	connections := make(map[string]interface{})
-
-	for {
-		var keys []string
-		pattern := fmt.Sprintf("%s:*", account)
-		if keys, cursor, err = rcl.RedisConnection.Scan(cursor, pattern, 50).Result(); err != nil {
-			// FIXME: log error, return an error
-			return connectionsPerAccount
-		}
-
-		if len(keys) > 0 {
-			fmt.Printf("found %d keys\n", len(keys))
-		}
-
-		for _, key := range keys {
-			connections[key] = nil
-		}
-
-		if cursor == 0 {
-			break
-		}
+	accountConnections, err := controller.GetRedisConnectionsByAccount(rcl.Client, account)
+	if err != nil {
+		// FIXME: Update connectionlocator interface methods to return error
+		logger.Log.Warnf("Error during lookup for account: %s", account)
+		return nil
 	}
 
-	for key, _ := range connections {
-		s := strings.Split(key, ":")
-		proxy := rcl.GetConnection(s[0], s[1])
-		connectionsPerAccount[key] = proxy
+	for _, conn := range accountConnections {
+		s := strings.Split(conn, ":")
+		nodeID := s[0]
+		proxy := rcl.GetConnection(account, nodeID)
+		// FIXME: Is this a good key?
+		connectionsPerAccount[account+":"+nodeID] = proxy
```

Comment:
I think this key breaks the interface.  In this method, we already know the account because that is what the caller passed in.  The key here should be the nodeID.

Also, I think the logic to determine the nodeID belongs in the redis_connector.go file now.  Notice that we've leaked the redis key details (the `strings.Split(conn, ":")`) into this file from the redis_connector.go file.  I think the GetRedisConnectionsByAccount() method should be modified to return a map[string]string.  They key would be the nodeID, the value would be the podname.  In the future, I suspect we will really need to return more metadata about the connection than just the podname/hostname but we can cross that bridge when we get there.

Reply:
Yeah good call. Will update